### PR TITLE
CORE-4875 Prevent creating multiple public sandboxes

### PR DESCRIPTION
**Pull request by mbrkic-r3**

### CODE REVIEW

File: libs/sandbox-internal/src/main/kotlin/net/corda/sandbox/internal/SandboxServiceImpl.kt [hunk 1]
```diff
@@ -55,6 +55,7 @@ internal class SandboxServiceImpl @Activate constructor(
     private val logger = loggerFor<SandboxServiceImpl>()
 
     override fun createPublicSandbox(publicBundles: Iterable<Bundle>, privateBundles: Iterable<Bundle>) {
+        check(publicSandboxes.isEmpty()) { "Public sandbox was already created" }
```
Comment (barrylapthorn) [inline libs/sandbox-internal/src/main/kotlin/net/corda/sandbox/internal/SandboxServiceImpl.kt]: I don't think we want to throw if there's a second attempt, I think we just want to log (warn?) and ignore it - in the combined worker we would have two or more calls to this at some point and the exception would halt execution wouldn't it?

Comment (mbrkic-r3) [reply libs/sandbox-internal/src/main/kotlin/net/corda/sandbox/internal/SandboxServiceImpl.kt]: OK, I'll change to warning (current version of combined worker doesn't have any issue, but not sure if other types of sandboxes are already being initialised or not)

Comment (driessamyn) [reply libs/sandbox-internal/src/main/kotlin/net/corda/sandbox/internal/SandboxServiceImpl.kt]: I think we said we should throw if it gets called twice with different input, but it's ok to warn if not.

Comment (mbrkic-r3) [reply libs/sandbox-internal/src/main/kotlin/net/corda/sandbox/internal/SandboxServiceImpl.kt]: Changed

**Review by barrylapthorn — APPROVED** [review]

**Pull request merged**